### ISSUE TITLE
`init` and attempting to upload new package from workflow.

### ISSUE THREAD

**Issue by yaakovschectman** [microsoft/setup-msstore-cli#6]
On the step:

      - name: CLI init
        run: msstore init . -n ${{ secrets.PUBLISHER_DISPLAY_NAME }} --package --publish

The workflow fails with the error message `Unhandled exception: System.NotSupportedException: Cannot show selection prompt since the current terminal isn't interactive.`

My goal is to build and package the application and upload a submission to the dev center, and it appeared that `init` is the closest command to do so, though I am not sure if it selects any particular app/submission to upload the built package to.

**Comment by azchohfi** [comment]
Hello @yaakovschectman.
Just to make sure I have full context, for what type of application is this for? (Flutter, React Native, Electron, WinUI, etc).
The `init` command is probably not the one you need here. You can call the `init` only once in your development computer. The CLI will connect with Partner Center, list all the applications in your account, and let you interactively select which application you want to configure your project. Then, it will detect the project type and properly configure your project. That command is not supposed to be called from CI/CD environments. What you want is probably a `package` command with the `--publish` option.
Does that make sense?
3 steps:
1) Init (once, in the dev box).
2) Package (in CI/CD), it will build and generate the MSIX.
3) Publish (in CI/CD), it will upload the MSIX to the store, to the right application that was selected in the `init` command.

**Comment by yaakovschectman** [comment]
I think so. I have run `msstore init` locally within the repository and pushed the changes it caused to the remote, then run package and publish in the workflow. Is there something that must be done in order to create/select a submission? `msstore publish` produces the below error:

```
Creating new Submission
💥 Error while creating submission. Please try again.
Could not retrieve submission. Please try again.
```

**Comment by azchohfi** [comment]
Is this the very first submission of the application, or is this an update?
If it is an update, what is the ApplicationCategory of this app?

**Comment by azchohfi** [comment]
Important to notice that if this is the first submission of an application, it needs to happen from an interactive console, as there are many missing information for the application, such as no category selected, for example.

**Comment by yaakovschectman** [comment]
It is an update, and it is listed as "Type: Game"

**Comment by yaakovschectman** [comment]
Hello, just want to check with you if you've any thoughts on the error message. I'm really not sure how to debug this one. Thanks.

**Comment by azchohfi** [comment]
It took me a long time to find this issue, but I believe we've fixed it.
Will ship a new version shortly.

**Comment by azchohfi** [comment]
Ok, new release is out on GitHub Actions.
Just run your build again, as it should pick up the latest version now.

**Comment by yaakovschectman** [comment]
Thank you for the update. I am currently running into the below error upon the publish command:

```
│ InvalidParameterValue │ You don’t have packages that support all of the      │
│                       │ device families you’ve selected in Device family     │
│                       │ availability. Upload a package that supports each    │
│                       │ device family you’ve selected, or uncheck the box    │
│                       │ for any device families that this submission doesn’t │
│                       │ support. Device families: Holographic                |
```

If this is creating a new submission, where is it getting the selected available families from? On the MS dev center, it looks like one can only select device families on a submission that's already started. Is there a CLI argument to specify the device families?

**Comment by azchohfi** [comment]
This is a separate issue that we are aware of, and we are already fixing it.
The CLI right now uses the same families for every type of project, but this should not be the case. It should use what previous submissions are using and have default per project type (UWP vs Win32 implies some of these, so different defaults).
If the submission is failed, you should be able to just change it and re-submit, without the CLI.

**Comment by azchohfi** [comment]
@yaakovschectman v0.1.23 should have fixed this issue. Let me know if you are still having problems.
